include/z64bgcheck.h

```diff
@@ -52,10 +52,26 @@ typedef struct {
 } CollisionPoly; // size = 0x10
 
 typedef struct {
-    /* 0x00 */ u16 cameraSType;
-    /* 0x02 */ s16 numCameras;
-    /* 0x04 */ Vec3s* camPosData;
-} CamData;
+    /* 0x0 */ u16 setting; // camera setting described by CameraSettingType enum
+    /* 0x2 */ s16 numData; // The total count of Vec3s data in the collision
+    /* 0x4 */ Vec3s* data; // See `SubBgCamData`
```

Comment:
`dataCount` would be preferable I think. I read this as "number data" and was very confused.
Even just `count` would work, since I think it would be implied that its the count for the data referenced by `BgCamData`. See for example `Path`, with `count` for the number of points

In that same vein, should it be `BgCamDataList`, since its a list of vec3s data? 
I am not fully following the connection between `BgCamData` and `SubBgCamData`, and how it knows what a certain vector is being used for

Reply:
I always viewed `count` as a number that can change, while `num` more representing a total (at least in the conventions of decomp). I was even recently thinking about changing `count` to `numPoints` in MM to follow convention more. Unless we want to move away from that line of thinking.

In terms of the relationship between `SubBgCamData` and `BgCamData`, I had an overly verbose description of the relationship between the two which you can see here:
https://github.com/engineer124/oot/blob/21b7133285a66408442ca81b1b538fe0a547e74b/include/z64bgcheck.h#L66-L97
Perhaps that'll help clarify and we can find a better way to document this.

I'm also open of using `BgCamDataList` in some way, but not exactly sure how. Could you clarify what you were thinking with that?

Reply:
i personally never viewed `count` as being dynamic or static. imo its just a more concise way of saying its the total amount of something. But others can weigh in on that as well

Reply:
`dataLength` ?

Reply:
In regards to `bgCamDataList`, that seems to be already used for the list of `bgCamData` in the scene (BgCamData[]), so also using it for the second list of Vec3s `bgCamData.data` (Vec3s[]) might be confusing

Reply:
`length`/`dataLength` is generic enough that I think it's fine to reuse it. Is there really a `bgCamDataList` length? Where?

Reply:
i think that response was to a point i raised earlier, unrelated to num/length/count

I had said that it seems like calling it a list might be better, cause it seems like a list of Vec3s.
But I was not aware of `bgCamDataList` already existing

Reply:
ah I see, mb

`BgCamData.data` is also only a list of Vec3s in the case of crawlspaces, I don't think the way things are right now are optimal but I'm waiting for crawlspace doc to make further comments and finalize how the struct works/looks

Reply:
Alright, I just synced crawlspaces to this PR, for reference:
https://github.com/engineer124/oot/compare/bgCamData...engineer124:crawlspaces?expand=1

Reply:
I would want to see a union but I guess it wouldn't behave correctly with ZAPD